### DISCUSSION TITLE
RNA velocity comparison between two samples

### DISCUSSION THREAD

**Discussion by rmurad-sbp** [theislab/scvelo#793] [Q&A]
Dear scvelo community,

We are trying to compare RNA velocities of two samples (KO vs. WT, please see below). Our progenitor cells are to the right of the UMAP (clusters 0, 10, and 11), intermediate states are in the middle (clusters 9, 1-4), while terminally differentiated states are clusters labeled 5, 6, and 8. From our scRNA-seq analysis and other data, we know that KO leads to fewer fully differentiated cells (cluster 5, 6, and 8). We noticed that in the KO RNA velocity plot within cluster 5, 6, and 8, there are fewer and thinner RNA velocity streamlines as compared to WT. 

Is it valid to deduce that KO RNA velocity shows decrease in differentiation potential of progenitors and intermediate states into fully differentiated states? Thank you in advance for your help/comments!

![image](https://user-images.githubusercontent.com/72945306/147787352-8efb6cbf-6f28-4659-b4fa-529daee2708c.png)

**Comment by ftucos** [comment]
You can get a better visual quantification by plotting directly velocity length (a proxy for relative speed in the transition) and confidence https://scvelo.readthedocs.io/VelocityBasics/#Speed-and-coherence.
I would also try to plot the embedding densities across a pseudo time axis for the lineage of your interest.

**Reply by rmurad-sbp** [reply]
Thank you! this was exactly what I was looking for.

**Comment by hyjforesight** [comment]
Hello @rmurad-sbp 
The UMAP embedding is beautiful. Could you please tell us how you make the same embedding between 2 samples and do scVelo on them?
Thanks!
Best,
YJ

**Reply by rmurad-sbp** [reply]
Hi,

Before processing the scRNA-seq data, you need to compute spliced and unspliced reads for each cell. I used Velocyto. Then, I integrated the two datasets (i.e. KO and WT) using Seurat CCA but using scTransform method:

https://satijalab.org/seurat/articles/integration_introduction.html#performing-integration-on-datasets-normalized-with-sctransform-1

After integration, I split the integrated Seurat object into KO and WT and ran scvelo on each individually. I hope this helps.

Best regards,

**Reply by hyjforesight** [reply]
Hello @rmurad-sbp ,
Thanks for the response.
We used to do a similar way as you did, but scvelo requires raw data as input. Please check this (https://github.com/theislab/scvelo/issues/775).

> In general, the data in adata.X should not be normalized when passing to scv.pp.filter_and_normalize. I suggest rerunning the pipeline with the raw counts in adata.X

 I think `adata.X` was normalized by scTransform before putting into the `scv.pp.filter_and_normalize`.

**Reply by rmurad-sbp** [reply]
Thank you for pointing this out. I did input SCT normalized data to scvelo with the exception of spliced and unspliced counts. At the scv.pp.filter_and_normalize stage, scvelo did give the message that the data was not normalized (spliced and unspliced counts were normalized only). I did require this, since I wanted the velocity streamlines to be superimposed on my already normalized and clustered data.

**Reply by kshirlekar** [reply]
Hello @rmurad-sbp! The approach described here is something I also used for my experiment with 2 conditions (Control and Treated). Even though I analyzed them separately, I would like to obtain the 2 embedding graphs next to each other using the same axes. Only for visualization purposes, I merged the two objects. Does scvelo have an argument like 'split.by' in Seurat to split the UMAPS as per the conditions? How did you obtain the plot above?
